tfx/dsl/component/experimental/decorators_test.py

```diff
@@ -93,6 +100,35 @@ def _injector_1(
   assert bar == 'secret'
   return {'a': 10, 'b': 22, 'c': 'unicode', 'd': b'bytes'}
 
+@component
+def _injector_1_with_typed_dict(
+        foo: Parameter[int], bar: Parameter[str]) -> TypedDict(
+            "InjectorOutput", a=int, b=int, c=str, d=bytes):
+  assert foo == 9
+  assert bar == 'secret'
+  return {'a': 10, 'b': 22, 'c': 'unicode', 'd': b'bytes'}
+
+class InjectorOutput(TypedDict):
+  a: int
+  b: int
+  c: str
+  d: bytes
+
+@component
+def _injector_1_with_custom_typed_dict(
+        foo: Parameter[int], bar: Parameter[str]) -> InjectorOutput:
+  assert foo == 9
+  assert bar == 'secret'
+  return {'a': 10, 'b': 22, 'c': 'unicode', 'd': b'bytes'}
+
+
+@component
+def _injector_1_with_custom_typed_dict_alter_syntax(
```

Comment:
This function is not used. Apparently, it has the same functionality of the function above, and we are not aiming to test the TypedDict itself, so I assume it can be safely erased.

Reply:
> This function is not used. Apparently, it has the same functionality as the function above.
True
>  and we are not aiming to test the TypedDict itself
While I understand where you're coming from, I wanted to test that intuitive ways to define annotations via `TypedDict` do work. I prefer to use this function instead (or escape the linting problem) as opposed to deleting the test. Are you fine with that or have any further suggestions?